src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts

```diff
@@ -43,9 +47,24 @@ export class ThemingStylesMigration extends Migration<ComponentMigrator[], Schem
       return m.styles.isLegacyMixin(this.namespace, atRule);
     });
     if (migrator) {
+      // Check to remove possible duplicate button theme mixins if migrating
+      // both button and snack bar since both include them
+      if (this.buttonMigrator && this.snackBarMigrator) {
+        if (migrator.styles.component === 'button') {
+          this.removeButtonThemesFromMigrator(this.snackBarMigrator);
+        } else if (migrator.styles.component === 'snack-bar') {
+          this.removeButtonThemesFromMigrator(this.buttonMigrator);
+        }
+      }
```

Comment:
Would it be easier to have something like:

```ts
mixinChanges = [
    {
      old: 'snack-bar-theme',
      new: [
        'mdc-snack-bar-theme',
        'mdc-snack-bar-typography',
        skipOnConflict('mdc-button-theme', ['button']),
        skipOnConflict('mdc-button-typography', ['button'])
      ],
    },
  ];
```

where `skipOnConflict` adds some special metadata to skip certain additions. Just an idea though. It's a little unfortunate that we need to encode migration-specific details here into the rather generic `theming-styles` rule

Reply:
@devversion I ended up adding this extra check for duplicates as a boolean in the StyleMigrator class that defaults to false but is true for snack-bar